packages/firestore/src/platform_node/grpc_connection.ts

```diff
@@ -142,7 +103,27 @@ export class GrpcConnection implements Connection {
     const stub = this.ensureActiveStub(token);
     const rpc = stub[rpcMethod];
     assert(rpc != null, 'Unknown RPC: ' + rpcName);
-    return rpc.bind(stub);
+
+    const metadata = new grpc.Metadata();
+    if (token) {
+      for (const header in token.authHeaders) {
+        if (token.authHeaders.hasOwnProperty(header)) {
+          metadata.set(header, token.authHeaders[header]);
+        }
+      }
+    }
+    metadata.set('x-goog-api-client', X_GOOG_API_CLIENT_VALUE);
+    // This header is used to improve routing and project isolation by the
+    // backend.
+    metadata.set(
+      'google-cloud-resource-prefix',
+      `projects/${this.databaseInfo.databaseId.projectId}/` +
+        `databases/${this.databaseInfo.databaseId.database}`
+    );
```

Comment:
FWIW, I think it's probably worth keeping this metadata creation in a helper.  `createMetadata(this.databaseInfo, token)` or whatever.

Reply:
Done